CONTRIBUTING.md

```diff
@@ -43,3 +43,10 @@ have LFS installed in your environment and run the following command:
 ```shell
 git lfs install
 ```
+
+### Links
+
+Use relative paths, e.g. `../path-to-file.md`, to link to other pages within the documentation. Relative links that
+include the `.md` file extension are checked during build, and broken links will cause builds to fail. This verification
+is not provided for absolute or external links. When linking to an index page, reference it explicitly, e.g.
+`path/to/index.md`.
```

Comment:
Circling back on https://github.com/kolenaIO/kolena/pull/72#issuecomment-1589218204, it appears that MkDocs does warn (and therefore fail on `--strict`) on broken internal links — but only relative links with `.md` file extensions. Absolute references and relative references without file extension (both of which work when deployed) have no such verification.

Added this section to get this behavior down in writing.